packages/opentelemetry-exporter-zipkin/src/zipkin.ts

```diff
@@ -34,19 +35,27 @@ export class ZipkinExporter implements SpanExporter {
   private readonly DEFAULT_SERVICE_NAME = 'OpenTelemetry Service';
   private readonly _statusCodeTagName: string;
   private readonly _statusDescriptionTagName: string;
+  private _urlStr: string;
   private _send: zipkinTypes.SendFunction;
+  private _getHeaders: zipkinTypes.GetHeaders | undefined;
   private _serviceName?: string;
   private _isShutdown: boolean;
   private _sendingPromises: Promise<unknown>[] = [];
 
   constructor(config: zipkinTypes.ExporterConfig = {}) {
-    const urlStr = config.url || ZipkinExporter.DEFAULT_URL;
-    this._send = prepareSend(urlStr, config.headers);
+    this._urlStr = config.url || ZipkinExporter.DEFAULT_URL;
+    this._send = prepareSend(this._urlStr, config.headers);
```

Comment:
instead of having beforeSend, couldn't we just transform prepareSend to use a callback to get headers ? The callback being defined here like:
```ts
this._send = prepareSend(this._urlStr, () => {
  let additionalHeaders = {}
  if (typeof config.getHeadersBeforeSend === 'function') {
    additionalHeaders = config.getHeadersBeforeSend()
  }
  return Object.assign({}, config.headers, additionalHeaders)
});
```

I think it would be easier to understand, WDYT ?

Reply:
there is much more logic with this, in web if there are no headers it will use sendBeacon otherwise it will use xhr. having the interceptor it means it will be dynamic then - exactly the same way you would restart service and start again. Also for majority of users this headers assignment is run once only that's why it was moved to constructor instead of checking with every request. So if user doesn't define an extra interceptor it will not change how it works now, otherwise it will, but the whole flow will be the same. In your case it will have to check everytime for `getHeadersBeforeSend` no matter if it was define or not.